extractor/filesystem/os/dpkg/dpkg.go

```diff
@@ -329,8 +329,22 @@ func (e Extractor) ToPURL(i *extractor.Inventory) *purl.PackageURL {
 	if m.Architecture != "" {
 		q[purl.Arch] = m.Architecture
 	}
+
+	typePurl := ""
+
+	for _, location := range i.Locations {
+		if strings.Contains(location, "opkg") {
```

Comment:
Would be better to check for full equality with `usr/lib/opkg/status`
DPKG can place DPKG files anywhere in `/var/lib/dpkg/status.d` so you could have a DPKG file called `/var/lib/dpkg/status.d/okpg-foo` that would incorrectly get an OPKG PURL type.